File: backend/models/plotDataSet.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type PlotDataSetSeriesData struct {
	X string  `bson:"x" json:"x"`
	Y float64 `bson:"y" json:"y"`
}

type PlotDataSetSeries struct {
	Name string                  `bson:"name,omitempty" json:"name,omitempty"`
	Data []PlotDataSetSeriesData `bson:"data" json:"data"`
}

type PlotDataSet struct {
	Id        primitive.ObjectID  `bson:"_id" json:"id,omitempty"`
	Series    []PlotDataSetSeries `bson:"series" json:"series"`
	ProjectId primitive.ObjectID  `bson:"project_id" json:"projectId,omitempty"`
}

File: miner/models/plot_data_set.py
from typing import List, TypedDict


# according to apexcharts.js

class PlotDataSetSeriesData(TypedDict):
    x: str
    y: int


class PlotDataSetSeries(TypedDict):
    name: str
    data: List[PlotDataSetSeriesData]


class PlotDataSet(TypedDict):
    series: List[PlotDataSetSeries]
    project_id: str
